### PR TITLE
chore(ci): use unified Gitlab pipeline for APM libraries

### PR DESCRIPTION
**Pull request by randomanderson**
**What does this PR do?**
Converts the Gitlab workflow to use the unified pipeline.

By using the unified pipeline across all APM libraries, there is less maintenance hassle, fewer mistakes, and a more uniform creation of artifacts.

Additionally, this change creates OCI packages and lib inject images for every `dd-trace-py` build. Previously, only released versions to pypi were built and tested.

The changes can be summarized as follows:
* Removing github actions that are now taken over by the unified pipeline
* Removing Gitlab jobs taken over by the unified pipeline
* Adding a job to download artifacts from the github build
* Merging the download artifacts with dependencies from pypi

`.gitlab-ci.yml` is best reviewed in its final form rather than as a diff.

**How to test the change?**
The changes were tested manually. CI/CD workflows are notoriously hard to test. Final products the final lib-init image was run through automated onboarding tests. `dd-trace-py` doesn't have automated tests for OCI images

## Checklist
- [x] PR author has checked that all the criteria below are met
- The PR description includes an overview of the change
- The PR description articulates the motivation for the change
- The change includes tests OR the PR description describes a testing strategy
- The PR description notes risks associated with the change, if any
- Newly-added code is easy to change
- The change follows the [library release note guidelines](https://ddtrace.readthedocs.io/en/stable/releasenotes.html)
- The change includes or references documentation updates if necessary
- Backport labels are set (if [applicable](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting))

## Reviewer Checklist
- [x] Reviewer has checked that all the criteria below are met 
- Title is accurate
- All changes are related to the pull request's stated goal
- Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes
- Testing strategy adequately addresses listed risks
- Newly-added code is easy to change
- Release note makes sense to a user of the library
- If necessary, author has acknowledged and discussed the performance implications of this PR as reported in the benchmarks PR comment
- Backport labels are set in a manner that is consistent with the [release branch maintenance policy](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting)

### CODE REVIEW

File: .gitlab/prepare-oci-package.sh [hunk 1]
```diff
@@ -0,0 +1,32 @@
+#!/bin/bash
+
+if [ -n "$CI_COMMIT_TAG" ] && [ -z "$PYTHON_PACKAGE_VERSION" ]; then
+  PYTHON_PACKAGE_VERSION=${CI_COMMIT_TAG##v}
+fi
+
+if [ -z "$PYTHON_PACKAGE_VERSION" ]; then
+  # Get the version from the filename of the first wheel
+  # wheels have the form:
+  # ddtrace-2.11.0.dev41+g50bf57680-cp312-cp312-manylinux_2_17_aarch64.manylinux2014_aarch64.wh
```
Comment (brettlangdon) [inline .gitlab/prepare-oci-package.sh:10]:
```suggestion
  # ddtrace-2.11.0.dev41+g50bf57680-cp312-cp312-manylinux_2_17_aarch64.manylinux2014_aarch64.whl
```

File: lib-injection/dl_wheels.py [hunk 2]
```diff
@@ -37,7 +37,8 @@
 if pip_version < MIN_PIP_VERSION:
     print(
         "WARNING: using known incompatible version, %r, of pip. The minimum compatible pip version is %r"
-        % (pip_version, MIN_PIP_VERSION)
+        % (pip_version, MIN_PIP_VERSION),
+        flush=True
```
Comment (brettlangdon) [inline lib-injection/dl_wheels.py:41]: > You can skip buffering for a whole python process using python -u or by setting the environment variable PYTHONUNBUFFERED.

**Review by brettlangdon — APPROVED** [review]
very minor, non-blocking, nits, but otherwise lgtm

thanks!

**Comment by brettlangdon** [comment]
you'll need to run `hatch run lint:fmt`, seems `lib-injection/dl_wheels.py` is failing linting.

**Review by gnufede — APPROVED** [review]

**Pull request merged**